### PR TITLE
Compound operation support

### PR DESCRIPTION
**Pull request by 333fred**
Added dataflow support for compound operations. I also added a few missing BinaryOperator tests for VB. @dotnet/roslyn-compiler and @dotnet/analyzer-ioperation for reviews.

### CODE REVIEW

File: src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ICompoundAssignmentOperation.vb [hunk 1]
```diff
@@ -430,6 +430,836 @@ BC30452: Operator '+' is not defined for types 'Module1.C' and 'Integer'.
...
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)>
+        <Fact()>
+        Public Sub CompoundAssignment_02()
+            Dim source = <![CDATA[
+Imports System
+Public Class C
+    Public Sub M(a As Integer(), b As Integer?, c As Integer)'BIND:"Public Sub M(a As Integer(), b As Integer?, c As Integer)"
+        a(0) += If(b, c)
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ICompoundAssignmentOperation.vb]: `-=`?

**Review by cston — APPROVED** [review]

File: src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ICompoundAssignmentOperation.vb (hunk 1 above)
Comment (333fred) [reply src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ICompoundAssignmentOperation.vb]: Fixed _02

**Comment by 333fred** [comment]
Ping @AlekseyTs and @dotnet/analyzer-ioperation .

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 3)

**Review by mavasani — APPROVED** [review]

**Pull request merged**